CHANGES: Fixed content and tool use both being present in API responses

Fried-Squid opened the pull request:
Fixes issue #134. May fix issue #109 but I have not tested deep research as it appears to require direct OpenAI API and we are using local models through LiteLLM.

Please test to ensure proper funtionality as this may be a breaking change

PhungVanDuy commented:
@khoangothe, please review this PR, which is essentially the same as my PR to fix the OpenAI endpoint.

khoangothe commented:
Hi @Fried-Squid, thank you for spotting the issue and submitting a PR! Would you mind removing all the debug breakpoints and fix the indent? Thank you so for the PR again!

khoangothe commented on src/ii_agent/llm/openai.py:
```diff
@@ -94,78 +94,80 @@ def generate(
+        print(f"OpenAI messages: {openai_messages}") 
+        
+        for idx, message_list in enumerate(messages):# Debug remove before PR
+                internal_message = message_list[0]  # Get the first message in the list
```
Indent twice here

Fried-Squid commented:
Fixed the indent, thanks for letting me know. I will update PR!

@khoangothe I havent had much time to look around the codebase as was just a quick fix in my head but I was going through the diff:

```py
if len(message_list) > 1:
    raise ValueError("Only one entry per message supported for openai")
internal_message = message_list[0]
```
This I replaced with a warning as changing to both tool_use and content means there can be multiple entries per message.
However, flattening out the message loop by iterating into internal_message (i.e. `for internal_message in message_list`) results in the OpenAI call in the next turn always returning None.

I'm not sure if this is an issue with my setup of the model as it uses local models and proxies to send the requests, or if the lack of for loop actually matters as the current solution of dropping message_list[1:] seems to have no effect on the core agent logic but I might be missing something.

khoangothe commented:
Hi @Fried-Squid, I was able to test and your PR runs well! Would you mind removing that print statement before I can merge the pull request
